pkgs/top-level/all-packages.nix

```diff
@@ -1334,19 +1334,21 @@ in
 
   cue2pops = callPackage ../tools/cd-dvd/cue2pops { };
 
-  cabal2nix = haskell.lib.overrideCabal (haskell.lib.generateOptparseApplicativeCompletion "cabal2nix" haskell.packages.ghc881.cabal2nix) (drv: {
-    isLibrary = false;
-    enableSharedExecutables = false;
-    executableToolDepends = (drv.executableToolDepends or []) ++ [ makeWrapper ];
-    postInstall = ''
-      exe=$out/libexec/${drv.pname}-${drv.version}/${drv.pname}
-      install -D $out/bin/${drv.pname} $exe
-      rm -rf $out/{bin,lib,share}
-      makeWrapper $exe $out/bin/${drv.pname} \
+  cabal2nix-unwrapped = haskell.lib.justStaticExecutables (haskell.lib.generateOptparseApplicativeCompletion "cabal2nix" haskellPackages.cabal2nix);
+
+  cabal2nix = stdenv.mkDerivation {
+    name = cabal2nix-unwrapped.name;
+    buildInputs = [ makeWrapper ];
+    meta = { inherit (cabal2nix-unwrapped.meta) name homepage license description platforms maintainers;};
+    buildCommand = ''
+      mkdir -p $out/bin
+      ln -s ${cabal2nix-unwrapped}/share $out
+      ln -s ${cabal2nix-unwrapped}/bin/hackage2nix $out/bin
+      makeWrapper ${cabal2nix-unwrapped}/bin/cabal2nix $out/bin/cabal2nix \
         --prefix PATH ":" "${nix}/bin" \
         --prefix PATH ":" "${nix-prefetch-scripts}/bin"
```

Comment:
I recommend this instead:
```
  --prefix PATH : "${lib.makeBinPath [ nix nix-prefetch-scripts ]}"
```

`makeBinPath` takes care of appending `/bin` and using the `bin` output if necessary

Reply:
Oh, that's cool! I'll do that right away.